mod/multi_file_upload/actions/file/upload.php

```diff
@@ -177,58 +176,79 @@
 			'subject_guid' => elgg_get_logged_in_user_guid(),
 			'object_guid' => $file->guid,
 		));
+
 	} else {
 		// failed to save file object - nothing we can do about this
 		$error = elgg_echo("file:uploadfailed");
 		register_error($error);
 
-		//log errors into session variable to display
+		// log errors into session variable to display
 		$failedMessage .= '<li>-'.$_FILES['upload']['name'][$i].'</li>';
 		$_SESSION['multi_file_upload_fail'] = elgg_echo('multi_upload:failed', array($failedMessage));
 		continue;
 	}
-
 }
 
-	$container = get_entity($container_guid);
-	if (elgg_instanceof($container, 'group')) {
-		error_log(json_encode([
-			'forwardURL' => [
-				"file/group/$container->guid/all"
-			],
-		]));
-		echo json_encode([
-			'forwardURL' => [
-				"file/group/$container->guid/all"
-			],
-			'count' => [
-				$container_guid
-			],
-			'name' => [
-				$_FILES['upload']['name'][0]
-			]
-		]);
-		forward("file/group/$container->guid/all");
-	} else {
-		error_log(json_encode([
-			'forwardURL' => [
-				"file/owner/$container->username"
-			],
-		]));
-		echo json_encode([
-			'forwardURL' => [
-				"file/owner/$container->username"
-			],
-			'count' => [
-				$container_guid
-			],
-			'name' => [
-				$_FILES['upload']['name'][0]
-			]
-		]);
-
-		forward("file/owner/$container->username");
+
+	/// check if the new notifications plugin is active, then trigger the plugin hook, otherwise, trigger event hook
+	/// if cp notification plugin is active, use that for notifications
+	if (elgg_is_active_plugin('cp_notifications')) {
+	
+		if ($folder_guid === 0) {
+			$forward_guid = $container_guid;
+			$group_guid = $container_guid;
+		} else {
+			$forward_guid = $folder_guid;
+			$group_guid = get_entity($file->container_guid)->getGUID();
+
+		}
+
+		$forward_entity = array(
+			'number_files_uploaded' => $number_of_files_uploaded,
+			'forward_guid' => $forward_guid,
+			'group_guid' => $group_guid,
+		);
+		elgg_trigger_event('multi_file_upload', 'object', $forward_entity);
+		elgg_register_event_handler('single_file_upload', 'object', 'cp_create_notification');
+		elgg_register_event_handler('single_zip_file_upload', 'object', 'cp_create_notification');
 	}
 
 
- ?>
+$container = get_entity($container_guid);
+if (elgg_instanceof($container, 'group')) {
+
+	/*error_log(json_encode([ 'forwardURL' => [ "file/group/$container->guid/all" ], ]));*/
```

Comment:
This is out of my scope and I am trying to do the least amount of change outside of my plugin, not sure if anyone needs this line to debug